src/app/functions/server/onerep.ts

```diff
@@ -265,6 +266,16 @@ export async function isEligible() {
     throw new Error("No session");
   }
 
+  const flag = await getFlag("FreeBrokerScan");
+
+  if (!flag.isEnabled) {
+    throw new Error("Flag is not enabled");
+  }
+
+  if (!flag.allowList?.includes(session.user.email)) {
+    throw new Error("User is not on allow list for feature");
+  }
+
```

Comment:
This is an example of how to gate a feature on a flag, `getFlag()` server function returns a `FeatureFlag` type.

If we need to do feature flags on the client side, we could expose a client function too (feature flags are exposed via an API it could fetch).

Reply:
If we expected _n_ (such as `allowList` here) to be very large (thousands or millions of entries), using `Set` instead of an array here would be advantageous. It might be faster still to build a SQL query that lets the database look it up and just return the answer vs. pulling all the data and doing it in TS (but we'd probably want to have a few tables with indexes instead of a column with array types as the schema is now).

I don't think it's likely we'll see those kind of numbers to warrant the more complex schema at this time. Using `Set` might be both more efficient for lookups and just as convenient though so we should consider using it here.

Reply:
I agree, we really shouldn't have more than a few tens or even hundreds of people on the allow list. Anything more than would probably require more sophisticated admin controls (groups, different policies etc)